docs/skip-test-stages.md

```diff
@@ -0,0 +1,77 @@
+
+Making test stages optional is useful in scenarios like:
+- reusing an existing cluster. For example, the prepare stage can be skipped because the cluster is already populated with data
+- focusing on specific stages of a test configuration. For example, Nemeses execution can be skipped if disruptions are not needed for a particular scenario
+- resource optimization. Skipping unnecessary steps can reduce resource consumption and speed up test execution, which is often useful in a local development environment
+
+### Configuration
+
+The test configuration parameter `optional_test_stages` is used to specify which test stages are optional. It is a dictionary where keys are the names of test stages, and boolean values indicate whether the stage should be executed or skipped.
+The parameter can be set in a test configuration file or passed as an environment variable.
+
+#### 1. Using configuration file
+
+If `optional_test_stages` is to be defined in a test configuration, it can be included as a separate dedicated test configuration file, or as a part of existing configuration.
+
+Example of defining test stages to execute/skip in a configuration file:
+```yaml
+optional_test_stages:
+  prepare_write: false
+  main_load: true
+  nemesis: true
+  data_validation: true
+```
+
+#### 2. Using SCT_OPTIONAL_TEST_STAGES environment variable
+
+Alternatively, `optional_test_stages` parameter can be set using the SCT_OPTIONAL_TEST_STAGES environment variable:
+- setting the value as a JSON-formatted string:
+```bash
+export SCT_OPTIONAL_TEST_STAGES='{"optional_test_stages": {"prepare_write": false, "main_load": true, "nemesis": true, "data_validation": true}}'
+```
+- setting the value from existing configuration file, as a YAML-formatted string:
+```bash
+export SCT_OPTIONAL_TEST_STAGES=$(cat ./configurations/optional_test_stages.yaml)
+```
+
+### Example scenario of skipping test stages
+
+The following example demonstrates executing a longevity test with skipped nemesis stage execution:
+```bash
+# prepare a dedicated configuration file with only the 'optional_test_stages' parameter
+cat <<EOF > configurations/optional_test_stages.yaml
+optional_test_stages:
+  prepare_write: true
+  main_load: true
+  nemesis: false
+  data_validation: true
+EOF
+
+export SCT_SCYLLA_VERSION=6.1
+
+# run longevity test, passing the newly created configuration file
+hydra run-test longevity_test.LongevityTest.test_custom_time --backend aws --config configurations/nemesis/longevity-5gb-1h-nemesis.yaml --config configurations/network_config/test_communication_public.yaml --config configurations/optional_test_stages.yaml
+
+# or use environment variable to pass the 'optional_test_stages' parameter to the test
+export SCT_OPTIONAL_TEST_STAGES=$(cat ./configurations/optional_test_stages.yaml)
```

Comment:
this doesn't look like working... shouldn't be single quoted? Also, in examples json is provided while here it's going to be yaml (but maybe it works, I didn't try).

Reply:
> in examples json is provided while here it's going to be yaml

sct_config.py::dict_or_str function was updated to be more robust, so it can consume json and yaml formatted strings (just using ast.literal_eval, as it was previously, was failing even on some dumb case, when e.g. bollean is provded as 'true', but not 'True').

>this doesn't look like working... shouldn't be single quoted?

it does work, e.g.:
```
❯ export SCT_OPTIONAL_TEST_STAGES=$(cat ./configurations/optional_test_stages.yaml)
❯ echo $SCT_OPTIONAL_TEST_STAGES
optional_test_stages:
  prepare_write: true
  wait_total_space_used: true
```
But you helped to find the issue in this guide - the env. var shouldn't include `optional_test_stages` key itself, only it's value. I'll update this.

Then, for setting yaml-formatted string into env. variable it would require a command like:
```
❯ export SCT_OPTIONAL_TEST_STAGES=$( { read; cat; } < ./configurations/optional_test_stages.yaml )
❯ echo $SCT_OPTIONAL_TEST_STAGES
  prepare_write: true
  wait_total_space_used: true
  main_load: true
  post_test_load: true
  nemesis: true
  data_validation: true
```

Do you think we need to provide this more complicated example? Or providing only example of setting the env variable from json-formatted string is enough?

Reply:
Updated doc page

Reply:
did you tried this ?
```
export SCT_OPTIONAL_TEST_STAGES.prepare_write='true'
```

Reply:
This way of setting hierarchical env. var is failing with `export: not valid in this context: SCT_SKIP_TEST_STAGES.nemesis`.
Seems like the only way to set structured data in an env. var is to serialize it into json/yaml strings when setting.